pkg/controller/vitessshard/reconcile_rollout.go

```diff
@@ -56,35 +56,36 @@ func (r *ReconcileVitessShard) reconcileRollout(ctx context.Context, vts *planet
 	if tabletKey == "" {
 		// If we have no more scheduled tablets, uncascade the shard.
 		if err := r.uncascadeShard(ctx, vts); err != nil {
-			r.recorder.Eventf(vts, v1.EventTypeWarning, "UncascadeFailed", "Failed to mark cascading shard rollout as complete: %v", err)
+			r.recorder.Eventf(vts, corev1.EventTypeWarning, "UncascadeFailed", "Failed to mark cascading shard rollout as complete: %v", err)
 			return resultBuilder.Error(err)
 		}
 
-		r.recorder.Eventf(vts, v1.EventTypeNormal, "RollingRestartComplete", "Cascading rollout of tablets is complete.")
+		r.recorder.Eventf(vts, corev1.EventTypeNormal, "RollingRestartComplete", "Cascading rollout of tablets is complete.")
 		return resultBuilder.Result()
 	}
 
 	masterEligibleTablets := vts.Spec.MasterEligibleTabletCount()
 	deletePod := false
 	tabletType := pod.Labels[planetscalev2.TabletTypeLabel]
 	// These two conditions guarantee that the tablet is a lone master.
-	if masterEligibleTablets == 1 && tabletType == string(planetscalev2.ReplicaPoolType) {
+	if masterEligibleTablets < 2 &&
```

Comment:
I assume the 0 case never happens or that we don't disambiguate it?

Reply:
Originally I had written it as `masterEligibleTablets == 1` because I thought 0 couldn't happen. After all, how could the total be 0 if we are looking at a master-eligible Pod right now?

I later realized, `masterEligibleTablets` tells us the desired number of tablets, whereas we might be looking at a Pod that hasn't been reconciled yet. So I decided we should handle the `0` case too.